### PR TITLE
docs(lineage): Updating Lineage feature guide

### PR DESCRIPTION
**Pull request by maggiehays**
Expands on manual lineage guide to include details around managing lineage edges via API; simplifies sidebar navigation to remove unnecessary subdirectory in `Enrich Metadata` section

## Checklist
- [x] The PR conforms to DataHub's [Contributing Guideline](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable). If a new feature has been added a Usage Guide has been added for the same.
- [ ] For any breaking change/potential downtime/deprecation/big changes an entry has been made in [Updating DataHub](https://github.com/datahub-project/datahub/blob/master/docs/how/updating-datahub.md)

### CODE REVIEW

**Review by jjoyce0510 — COMMENTED** [review]
This is awesome~! We have another Lineage feature guide written by Chris that focuses on UI lineage editing. Maybe we can merge this into that? (It was just merged)

File: docs/lineage/lineage-feature-guide.md [hunk 1]
```diff
@@ -100,8 +168,8 @@ Using the modal from this view will work the same as described above for editing
 
 ### GraphQL
 
-* [updateLineage](../../graphql/mutations.md#updatelineage)
```
Comment (chriscollins3456) [inline docs/lineage/lineage-feature-guide.md:103]: why remove this reference to `updateLineage`?

Comment (jjoyce0510) [reply docs/lineage/lineage-feature-guide.md:103]: +1

Comment (maggiehays) [reply docs/lineage/lineage-feature-guide.md:103]: Good catch - this was a mistake. Adding it back in

**Comment by maggiehays** [comment]
@chriscollins3456 @jjoyce0510 this is ready for final review. One thing to consider -- this PR removes `intro.md` which is our legacy entry point for lineage.. it was extremely low value, but it will result in a broken link. Should we think through redirects?

File: docs/lineage/lineage-feature-guide.md [hunk 2]
```diff
@@ -82,14 +88,76 @@ The other place that you can edit lineage for entities is from the Lineage Tab o
...
+
+### Using Dashboard-to-Chart Lineage
+
+This relationship model uses the dashboardInfo aspect of the Dashboard entity and models an explicit edge between a dashboard and a chart (such that charts can be attached to multiple dashboards).
+
+Here is a sample usage of this lineage:
+
+* [lineage_chart_dashboard.py](../../metadata-ingestion/examples/library/lineage_chart_dashboard.py) - emits the chart-to-dashboard lineage via REST as MetadataChangeProposalWrapper.
+
+### Using Chart-to-Dataset Lineage
```
Comment (jjoyce0510) [inline docs/lineage/lineage-feature-guide.md:145]: Yeah this is much better.

**Review by jjoyce0510 — APPROVED** [review]

**Pull request merged**